### DISCUSSION TITLE
Inherit tenant settings to site properties when creating a new Teamsite

### DISCUSSION THREAD

**Discussion by web265p3** [pnp/pnpcore#1325] [General]
When a new Teamsite (without a group) is created programmatically, the sharing options for external users are disabled if I use PnP.Core.SDK.

At our tenant level we have enabled sharing for new and existing guests. This is the preferred setting for all of our sites.

I know that I can change those settings like this:

```
var siteProperties = await context.GetSiteCollectionManager().GetSiteCollectionPropertiesAsync(new Uri("https://contoso.sharepoint.com/sites/sitetocheck"));

// Update site properties
siteProperties.SharingCapability = SharingCapabilities.ExternalUserAndGuestSharing;
```

But I do not want to set an explicit setting here. I want all sites to use the tenant level settings.

Is there any way to archive this? I do not see any setting like "Inherit" or something similar...

**Comment by jansenbe** [comment]
@web265p3 : not sure, I don't think that's possible today as I don't recall there's a 'creation option' we can specify at site creation time that would do this. I'll need to cross check to see if we've missed a property.

Why is the workaround not an option?

**Comment by web265p3** [comment]
The workaround is not an option, because if later we change our mind about this setting, we needed to go to ANY Teamsite to change this option back.
It would be far more handy to have the option on the tenant level - like its by default - and change it there and the changes would apply for all sites.
But otherwise I need to write a skript to undo those changes... also fine ;)

**Reply by jansenbe** [reply]
@web265p3 : just to confirm, when you create the site using the UI the settings are correctly applied/used?

**Reply by web265p3** [reply]
Correct, when I use the UI its like it should be.
